### PR TITLE
feat: adds test infrastructure to run acceptance tests on PRs

### PR DESCRIPTION
**Pull request by marcosuma**
## Description

Link to any related issue(s): https://jira.mongodb.org/browse/INTMDB-1034

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [X] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [X] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [X] I have added any necessary documentation (if appropriate)
- [X] I have run make fmt and formatted my code
- [ ] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.

### CODE REVIEW

File: .github/workflows/acceptance-tests.yml [hunk 1]
```diff
@@ -31,13 +31,16 @@ jobs:
       serverless: ${{ steps.filter.outputs.serverless }}
       network: ${{ steps.filter.outputs.network }}
       config: ${{ steps.filter.outputs.config }}
+      assume_role: ${{ steps.filter.outputs.config }}
```
Comment (lantoli) [inline .github/workflows/acceptance-tests.yml]:
```suggestion
      assume_role: ${{ steps.filter.outputs.assume_role }}
```

File: .github/workflows/acceptance-tests.yml [hunk 2]
```diff
@@ -31,13 +31,16 @@ jobs:
...
     steps:
     - uses: actions/checkout@v4
       if: ${{ github.event_name == 'push' ||  github.event_name == 'workflow_dispatch' || github.event_name == 'schedule' || inputs.parent-event-name == 'release' }}
     - uses: dorny/paths-filter@v2
       id: filter
       with:
         filters: |
+          assume_role:
+            - 'mongodbatlas/**provider**.go'
```
Comment (lantoli) [inline .github/workflows/acceptance-tests.yml:43]: this seems very broad and it's also matched e.g. in config, what list of files do we want to match here?

File: .github/workflows/acceptance-tests.yml [hunk 3]
```diff
@@ -103,6 +106,28 @@ jobs:
             - 'mongodbatlas/resource_mongodbatlas_team*.go'
             - 'mongodbatlas/resource_mongodbatlas_third_party_integration*.go'     
 
+  fetch-sts-assume-role-creds:
+    runs-on: ubuntu-latest
+    outputs:
```
Comment (lantoli) [inline .github/workflows/acceptance-tests.yml:111]: don we need something like this?
 needs: [ change-detection ]
    if: ${{ needs.change-detection.outputs.assume-role == 'true'

**Review by lantoli — APPROVED** [review]

File: .github/workflows/acceptance-tests.yml (hunk 1 above)
Comment (marcosuma) [reply .github/workflows/acceptance-tests.yml]: phew, thank you!

File: .github/workflows/acceptance-tests.yml (hunk 2 above)
Comment (marcosuma) [reply .github/workflows/acceptance-tests.yml:43]: I thought of making it broad because I'd thought it should run for all PR changes since it's an authentication type of test. Happy to change here later if we see it's not worth it

Comment (lantoli) [reply .github/workflows/acceptance-tests.yml:43]: in that case we may want to have it in the general "config" group instead of a separated asssume-role group.

Comment (marcosuma) [reply .github/workflows/acceptance-tests.yml:43]: let's see how it goes, I'll keep this comment into account to improve things. Let us just see how it works.

File: .github/workflows/acceptance-tests.yml [hunk 4]
```diff
@@ -424,3 +449,33 @@ jobs:
...
+          AWS_SESSION_TOKEN: ${{ needs.fetch-sts-assume-role-creds.outputs.AWS_SESSION_TOKEN }}
+          MONGODB_ATLAS_ORG_ID: ${{ vars.MONGODB_ATLAS_ORG_ID_CLOUD_DEV }}
+          MONGODB_ATLAS_BASE_URL: ${{ vars.MONGODB_ATLAS_BASE_URL }}
+          ACCTEST_TIMEOUT: ${{ vars.ACCTEST_TIMEOUT }}
+          TF_LOG: ${{ vars.LOG_LEVEL }}
+          TF_ACC: 1
+          PARALLEL_GO_TEST: 20
+          CI: true
+          TEST_REGEX: "^TestAccSTSAssumeRole"
+        run: make testacc
```
Comment (andreaangiolillo) [inline .github/workflows/acceptance-tests.yml:481]: @marcosuma This change is leaking our AWS credentials: 

https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/6728120776/job/18286966224#step:4:10. Please revert this change and recreate our AWS keys
